test/create.test.ts

```diff
@@ -591,3 +591,90 @@ describe('usage with effects', () => {
 		expect(result).toBeTypeOf('string');
 	});
 });
+
+describe('create Intersection', () => {
+	test('creates an Intersection', () => {
+		const result = createFixture(
+			z.object({
+				a: z.string(),
+			}).and(z.object({
+				b: z.number(),
+			}))
+		);
+		expect(result).toHaveProperty('a');
+		expect(result).toHaveProperty('b');
+	});
+
+	test('creates a nested intersection', () => {
+		const result = createFixture(
+			z.object({
+				str: z.string(),
+				nested: z.object({
+					num: z.number(),
+					date: z.date(),
+				}),
+			}).and(z.object({
+					str2: z.string(),
+					nested2: z.object({
+						num: z.number(),
+						date: z.date(),
+					}),
+				}),
+			));
+		expect(result).toBeTypeOf('object');
+		expect(result.str).toBeTypeOf('string');
+		expect(result.str).toContain('str');
+		expect(result.nested.num).toBeTypeOf('number');
+		expect(result.nested.date).toBeInstanceOf(Date);
+		expect(result.str2).toBeTypeOf('string');
+		expect(result.str2).toContain('str');
+		expect(result.nested2.num).toBeTypeOf('number');
+		expect(result.nested2.date).toBeInstanceOf(Date);
+	});
+
+	test("creates an intersection with zod's api", () => {
+		const BaseTeacher = z.object({ students: z.array(z.string()) });
+		const HasID = z.object({ id: z.string() });
+		const Teacher = z.intersection(BaseTeacher, HasID);
+
+		const result = createFixture(Teacher);
+		expect(result).toBeTypeOf('object');
+		expect(result.id).toBeTypeOf('string');
+		expect(result.id).toContain('id');
+		expect(result.students).toHaveLength(3);
+	});
+
+	test('creates intersection with optional value ', () => {
+		const SampleWithOptionalValueSchema = z.object({
+			name: z.string().optional(),
+		}).and(z.object({
+			modify: z.boolean().optional(),
+		}));
+		expect(() => {
+			createFixture(SampleWithOptionalValueSchema);
+		}).not.toThrow();
+	});
+
+	test('creates intersection with transform', () => {
```

Comment:
It seems like this test fails 👀

Reply:
Oh sorry, missed that after I added the type check. Should be good now!